app/queries/project/subscriptions_pledged_value_query.rb

```diff
@@ -0,0 +1,17 @@
+class Project::SubscriptionsPledgedValueQuery
+  attr_reader :subscriptions
+
+  def initialize(project)
+    @subscriptions = project.subscriptions.includes(:plan).with_paid_transactions
+  end
+
+  def self.call(project)
+    new(project).call
+  end
+
+  def call
+    subscriptions.inject(0) do |pledged_value, subscription|
+      pledged_value + (subscription.transactions.length * subscription.plan.formatted_amount)
```

Comment:
When you have `has_many` association, the worse counter is `length`, because it loads all the objects and then it calculates how many objects there are. You can use `count` or `size`.

`count` makes a `COUNT query` every time you call it;
`size` makes a `COUNT query` the first time you call it (will adapt your needs).

You can [see here](http://stackoverflow.com/questions/6083219/activerecord-size-vs-count)